src/rabbit_plugins.erl

```diff
@@ -245,19 +253,100 @@ prepare_plugins(Enabled) ->
 
+format_invalid_plugins(InvalidPlugins) ->
+    lists:flatten(["Failed to enable some plugins: \r\n"
+                   | [format_invalid_plugin(Plugin)
+                      || Plugin <- InvalidPlugins]]).
+
+format_invalid_plugin({Name, Errors}) ->
+    [io_lib:format("    ~p:~n", [Name])
+     | [format_invalid_plugin_error(Err) || Err <- Errors]].
+
+format_invalid_plugin_error({missing_dependency, Dep}) ->
+    io_lib:format("        Dependency is missing or invalid: ~p~n", [Dep]);
+format_invalid_plugin_error({version_mismatch, {Version, Required}}) ->
+    io_lib:format("        Broker version is invalid."
+                  " Current version: ~p Required: ~p~n", [Version, Required]);
+format_invalid_plugin_error({{version_mismatch, {Version, Required}}, Name}) ->
+    io_lib:format("        ~p plugin version is invalid."
+                  " Current version: ~p Required: ~p~n",
+                  [Name, Version, Required]);
+format_invalid_plugin_error(Err) ->
+    io_lib:format("        Unknown error ~p~n", [Err]).
+
+validate_plugins(Plugins) ->
+    application:load(rabbit),
+    RabbitVersion = RabbitVersion = case application:get_key(rabbit, vsn) of
+                                        undefined -> "0.0.0";
+                                        {ok, Val} -> Val
+                                    end,
+    validate_plugins(Plugins, RabbitVersion).
+
+validate_plugins(Plugins, RabbitVersion) ->
+    lists:foldl(
+        fun(#plugin{name = Name, 
+                    rabbitmq_versions = RabbitmqVersions,
+                    plugins_versions  = PluginsVersions} = Plugin,
+            {Plugins0, Errors}) ->
+            case version_support(RabbitVersion, RabbitmqVersions) of
+                {error, Err} -> {Plugins0, [{Name, [Err]} | Errors]};
+                ok           ->
+                    case check_plugins_versions(Plugins0, PluginsVersions) of
+                        ok           -> {[Plugin | Plugins0], Errors};
+                        {error, Err} -> {Plugins0, [{Name, Err} | Errors]}
+                    end
+            end
+        end,
+        {[],[]},
+        Plugins).
+
+check_plugins_versions(AllPlugins, RequiredVersions) ->
+    ExistingVersions = [{Name, Vsn}
+                        || #plugin{name = Name, version = Vsn} <- AllPlugins],
+    Problems = lists:foldl(
+        fun({Name, Versions}, Acc) ->
+            case proplists:get_value(Name, ExistingVersions) of
+                undefined -> [{missing_dependency, Name} | Acc];
+                Version   ->
+                    case version_support(Version, Versions) of
+                        {error, Err} -> [{Err, Name} | Acc];
+                        ok           -> Acc
+                    end
+            end
+        end,
+        [],
+        RequiredVersions),
+    case Problems of
+        [] -> ok;
+        _  -> {error, Problems}
+    end.
+
+version_support(_Version, []) -> ok;
+version_support(Version, ExpectedVersions) ->
```

Comment:
I'd recommend renaming this to `is_version_supported` or `check_version_compatibility`.